### PR TITLE
Refactor check_requirements.py

### PR DESCRIPTION
**Pull request by finagle29**
Using `setuptools`'s `pkg_resources` to find packages and verify their version has caused headaches in the past (see pytest erroring CI builds because `pkg_resources` thought its version was 0.0.0, using `conda install pyside2` with PypeIt v1.3 causes `pkg_resources` to complain that PySide2 is not installed, even though it can be imported).
Directly importing packages and checking their `__version__` attribute is more stable than relying on packages to correctly create and install a `.egg-info` or `.dist-info` metadata directory along with the package itself.

Additionally, this reduces the severity of a package not being found from an error to a warning, and allows an optional error/warning message to be specified in `requirements_imports.txt` (good for `shapely` and `pytest`, which are not required by all users). Note, however, that `requirements_imports.txt` cannot just be a copy of `requirements.txt` since some package names (looking at you `extension-helpers`) are not valid Python identifiers and the name of the module that is imported is different that what is in `requirements.txt` (or `PyYAML` vs `import yaml`).

Happy to hear suggestions/feedback on my particular implementation of this, especially on the `requirements_imports.txt` file (its format, filename, any other [optional] fields we want to put in it).

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -19,5 +19,4 @@ dependencies:
   - extension-helpers>=0.1
   - ginga>=3.0
   - linetools>=0.3
-  - PySide2>=5.0
```
Comment (tepickering) [inline environment.yml:22]: this is being removed from here and letting the dependencies that require it pull it in, correct?

**Review by tepickering — COMMENTED** [review]
looks good to me. i'm hoping this stuff will become somewhat obviated once dependency handling is integrated into `setup.cfg`, but probably still good to have around to help diagnose user environments.

**Review by jhennawi — COMMENTED** [review]
Hi Milan, 

Thanks for this PR. Can you please make sure that doc/installing.rst is up to date, specifically that the dependencies there match your changes to the requirements files and environment.yaml. Also, please double check that the installing from source instructions are up to date, and that this works via the Conda environment. I think @tepickering is working on a PR that tests this for different architectures. 

Also a more general point regarding check_requirements. Currently this is setup such that the following line is in pypeit/__init__.py 

from pypeit import check_requirements  # THIS IMPORT DOES THE CHECKING.  KEEP IT

This will then faults anytime you try to run pypeit if all the requirements are not installed. I don't think this is desirable or necessary. My opinion is that we should just be checking these requirements once upon installation via the python setup.py script or whatever script we choose for installation. But perhaps we can learn from how other packages deal with dependencies. @tepickering do you know how this is handled with astropy? I've never seen astropy crash nearly as much as Pyepit because of missing dependencies. 

Thanks,
Joe

File: environment.yml (hunk 1 above)
Comment (finagle29) [reply environment.yml:22]: PySide2 was actually duplicated in environment.yml (might have been my fault a while back oops) so I'm just removing the duplicated entry

Comment (profxj) [reply environment.yml:22]: Was probably my fault (via a merge).

**Comment by finagle29** [comment]
Good point about updating the docs Joe, I'll take care of that.
And regarding your second point, I'm more agnostic about whether we should check requirements only upon installation or every time PypeIt runs. Note that this PR decreasing the level from an error/exception to a warning means that PypeIt will run even if some requirements are not satisfied (though of course if something like scipy is missing then PypeIt will crash very quickly with `ModuleNotFoundError: no module named 'scipy'`).

```console
$ run_pypeit -h
[WARNING] :: Your pytest (version 5.4.3) is incompatible with PypeIt. Please update to version >= 6.0.0. pytest is only required for running tests
[WARNING] :: Package shapely not found. shapely is required only for KCWI
usage: run_pypeit [-h] [-v VERBOSITY] [-t] [-r REDUX_PATH] [-m] [-s] [-o] [-d DETECTOR] [-c] pypeit_file
...
```

**Comment by finagle29** [comment]
superseded by #1130, closing.

**Pull request closed without merging**